src/GoldMiner/Hook.java

```diff
@@ -0,0 +1,199 @@
+     * @return
+     */
+    public void updatePosition(double angle) {
+        if (hook.getX() >= 0 && hook.getX() <= maxX && hook.getY() <= maxY) {
+            hook.moveBy(moveX, moveY);
+        } else {
+            moveX = 0 - moveX;
+            moveY = 0 - moveY;
+            hook.moveBy(10 * moveX, 10 * moveY);
+        }
+
+        for (GraphicsObject oneMineral : mineralList) {
+            if (oneMineral.getY() >= 900) {
+                canvas.remove(oneMineral);
+                mineralList.remove(oneMineral);
+                break;
+            } else {
+                getMineral(oneMineral);
+            }
+        }
+
+    }
+
+    /**
+     * This method update the moveX and moveY of the hook by given velocity.
+     * 
+     * @param velocity
+     */
+    public void updateDirection(double velocity) {
+        moveX = velocity * Math.sin(angle);
+        moveY = velocity * Math.cos(angle);
+    }
+
+
+    /**
+     * This method takes angle in degrees as parameter, update the position of the aiming line.
+     * 
+     * @param angleInDegrees
+     */
+    public void updateAiming(double angleInDegrees) {
+        angle = Math.toRadians(angleInDegrees);
+        x2 = Math.sin(angle) * 40 + INITIAL_X;
+        y2 = Math.cos(angle) * 40 + INITIAL_Y;
+        hookaiming.setEndPosition(x2, y2);
+        canvas.draw();
+        canvas.pause(50);
+    }
+
+    /**
+     * If the disctance between the hook and the mineral is less than 200, the hook move back to the
+     * origin, and the mineral move with the hook.
+     * 
+     * @param g
+     * @return
+     */
+    public void getMineral(GraphicsObject mineral) {
+
+        double radius = mineral.getSize().getX() / 2;
+        velocity = 20 / radius;
+        // As the radius of the mineral increase, the velocity of the hook will decrease.
+
+        if (distance(mineral) <= Math.pow(radius, 2)) {
+            if (moveY >= 0) {
+                updateDirection(velocity);
+                moveX = 0 - moveX;
+                moveY = 0 - moveY;
+            }
+            hook.moveBy(moveX, moveY);
+            mineral.moveBy(2 * moveX, 2 * moveY);
+            // Remove the mineral when it get to the initial point of the hook.
+            if (mineral.getY() <= INITIAL_Y + 4) {
+                mineral.setPosition(1000, 1000);
+                // Change the velocity of the hook
+                if (mineral.equals(gold.diamond) || mineral.equals(gold.diamond1)
+                    || mineral.equals(gold.diamond2)) {
+                    // if the mineral is diamond
+                    score += 100;
+
+                } else if (mineral.getSize().getX() != mineral.getSize().getY()) {
+                    // if the mineral is stone
+                    score += radius;
+                }else {
+                    // if the mineral is gold
+                    score += radius * 3;
+                }
```

Comment:
Instead of using size and comparing with specific objects like this, I’d recommend making a `Mineral` class that knows its score.